File: HRMS/HRMS.Web/Modules/ServerTypes/Communication/SocialPostMediaColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SocialMediaType } from "./SocialMediaType";
import { SocialPostMediaRow } from "./SocialPostMediaRow";

export interface SocialPostMediaColumns {
    FileName: Column<SocialPostMediaRow>;
    MediaType: Column<SocialPostMediaRow>;
    DisplayOrder: Column<SocialPostMediaRow>;
}

export class SocialPostMediaColumns extends ColumnsBase<SocialPostMediaRow> {
    static readonly columnsKey = 'Communication.SocialPostMedia';
    static readonly Fields = fieldsProxy<SocialPostMediaColumns>();
}

[SocialMediaType]; // referenced types

File: HRMS/HRMS.Web/Modules/Communication/SocialPost/SocialPostMediaColumns.cs
using Serenity.ComponentModel;

namespace HRMS.Communication.Columns;

[ColumnsScript("Communication.SocialPostMedia")]
[BasedOnRow(typeof(SocialPostMediaRow), CheckNames = true)]
public class SocialPostMediaColumns
{
    [EditLink]
    public string FileName { get; set; }
    public SocialMediaType MediaType { get; set; }
    public int DisplayOrder { get; set; }
}
